sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/authentication.py

```diff
@@ -75,31 +84,33 @@ def __init__(
         self.audience = audience
         self.token_type = token_type
         self.get_token = get_token
-        self.expires_in = kwargs.pop("expires_in", AUTH_DEFAULT_EXPIRATION_SECONDS)
-        self.expires_on = kwargs.pop("expires_on", None)
+        self.expires_in = expires_in
+        self.expires_on = expires_on
 
     @staticmethod
-    def _set_expiry(expires_in, expires_on):
+    def _set_expiry(expires_in: Optional[float] = None, expires_on: Optional[float] = None) -> Tuple[float,float]:
         if not expires_on and not expires_in:
             raise ValueError("Must specify either 'expires_on' or 'expires_in'.")
-        if not expires_on:
+        if not expires_on and expires_in:
             expires_on = time.time() + expires_in
-        else:
+        elif expires_on and not expires_in:
             expires_in = expires_on - time.time()
             if expires_in < 1:
                 raise ValueError("Token has already expired.")
-        return expires_in, expires_on
+        return cast(float, expires_in), cast(float, expires_on)
```

Comment:
I had to cast here because mypy was complaining that `*` cant be applied on values that can be None. I think I can get around it with some local vars.